### PR TITLE
Call accounts

### PR DESCRIPTION
**Pull request by nagarajmanjunath**
* update web3 crate
* update readme

### CODE REVIEW

File: web3/src/contract_call.rs [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+	let sender = tx_details.clone().unwrap().0.sender;
+	let contract_address = tx_details.clone().unwrap().1.contract;
+	let gas_left = tx_details.clone().unwrap().1.gas_left;
+	let u1 = U256::zero();
+
+	let receipt = TxReceipt {
+		transaction_hash: params,
+		blockHash: bc.last_block_hash(),
+		from: sender,
+		to: H160::zero(),
```
Comment (b00f) [inline web3/src/contract_call.rs]: Is it correct? Receiver should have an address.

File: web3/src/lib.rs [hunk 2]
```diff
@@ -1,3 +1,33 @@
...
+pub mod rpc_service;
+pub mod http_common;
+pub mod contract_call;
+
+
+pub use jsonrpc_core::{Compatibility, Error, MetaIoHandler};
+pub use jsonrpc_http_server::Server;
+// pub use rpc_service::start_http;
+extern crate log;
+pub mod v1;
```
Comment (b00f) [inline web3/src/lib.rs]: I don't think we need v1 interfaces. Better have only one version of interfaces .

File: web3/src/v1/helpers/sign.rs [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+
+
+use std::cmp::min;
+
+use ethereum_types::U256;
+use jsonrpc_core::Error;
+use v1::helpers::request::CallRequest;
+
+// pub fn sign_call(request: CallRequest) -> Result<SignedTransaction, Error> {
```
Comment (b00f) [inline web3/src/v1/helpers/sign.rs]: I think this file is not necessary .

File: web3/src/v1/types/block_number.rs [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+// along with Parity Ethereum.  If not, see <http://www.gnu.org/licenses/>.
+
+use std::fmt;
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+use serde::de::{Error, Visitor, MapAccess};
+use ethereum_types::H256;
+
+/// Represents rpc api block number param.
+#[derive(Debug, PartialEq, Clone, Hash, Eq)]
+pub enum BlockNumber {
```
Comment (b00f) [inline web3/src/v1/types/block_number.rs]: Can we simplify it. I think we don't need to have block_number in web3 .

File: web3/src/v1/types/bytes.rs [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+}
+
+impl<'a> Deserialize<'a> for Bytes {
+	fn deserialize<D>(deserializer: D) -> Result<Bytes, D::Error>
+	where D: Deserializer<'a> {
+		deserializer.deserialize_any(BytesVisitor)
+	}
+}
+
+struct BytesVisitor;
```
Comment (b00f) [inline web3/src/v1/types/bytes.rs]: Can we omit it?  I checked we haven't used it in our code.

File: web3/src/v1/types/provenance.rs [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+use serde::{Serialize, Deserialize};
+/// RPC request origin
+#[derive(Clone, Debug, PartialEq, Eq, Hash, Serialize, Deserialize)]
+#[serde(deny_unknown_fields)]
+#[serde(rename_all = "kebab-case")]
+pub enum Origin {
+	/// RPC server (includes request origin)
+	Rpc(String),
+	/// IPC server (includes session hash)
+	Ipc(H256),
```
Comment (b00f) [inline web3/src/v1/types/provenance.rs]: Better we only keep RPC. I think this enum is extra.

File: web3/src/contract_call.rs (hunk 1 above)
Comment (nagarajmanjunath) [reply web3/src/contract_call.rs]: It's correct! bcoz its transaction receipt, which is generated manually.

File: web3/src/lib.rs (hunk 2 above)
Comment (nagarajmanjunath) [reply web3/src/lib.rs]: I agree it we don't want v1, but we are planning to update to 4.0, So I didn't modify it

File: web3/src/v1/helpers/sign.rs (hunk 3 above)
Comment (nagarajmanjunath) [reply web3/src/v1/helpers/sign.rs]: I was planning to keep all the call and create for smart contract related function in a separate file.

File: web3/src/v1/types/bytes.rs (hunk 5 above)
Comment (nagarajmanjunath) [reply web3/src/v1/types/bytes.rs]: yes can omit

File: web3/src/v1/types/provenance.rs (hunk 6 above)
Comment (nagarajmanjunath) [reply web3/src/v1/types/provenance.rs]: i ll update it!

File: web3/src/v1/types/block_number.rs (hunk 4 above)
Comment (nagarajmanjunath) [reply web3/src/v1/types/block_number.rs]: its using serialize and deserialise from ```crate:serde:``` we might keep it.

File: web3/src/contract_call.rs (hunk 1 above)
Comment (b00f) [reply web3/src/contract_call.rs]: > `to: H160::zero(),`

To address is nil?

File: web3/src/lib.rs (hunk 2 above)
Comment (b00f) [reply web3/src/lib.rs]: That part is not web3. Clean-up the code. Let have clean workspace.

File: web3/src/v1/helpers/sign.rs (hunk 3 above)
Comment (b00f) [reply web3/src/v1/helpers/sign.rs]: But we supposed to not sign message here. right?

**Comment by b00f** [comment]
@nagarajmanjunath  Please fix address these comments. Thanks

File: web3/src/contract_call.rs (hunk 1 above)
Comment (nagarajmanjunath) [reply web3/src/contract_call.rs]: yes, ```to``` will nill reason, a transaction receipt is sent to contract creator which is ```from:sender``` with contract address ```contractAddress: contract_address```.

File: web3/src/lib.rs (hunk 2 above)
Comment (nagarajmanjunath) [reply web3/src/lib.rs]: sure bro, I have update the code. please review it any changes update me.

File: web3/src/v1/helpers/sign.rs (hunk 3 above)
Comment (nagarajmanjunath) [reply web3/src/v1/helpers/sign.rs]: I have removed this function, I have sent a new pull request.

**Comment by nagarajmanjunath** [comment]
Hi bro, I have updated the code with latest changes and i have sent the pull request

**Pull request merged**